src/H5ES.c

```diff
@@ -235,6 +235,61 @@ H5ESget_op_counter(hid_t es_id, uint64_t *op_counter /*out*/)
     FUNC_LEAVE_API(ret_value)
 } /* end H5ESget_op_counter() */
 
+/*-------------------------------------------------------------------------
+ * Function:    H5ESget_requests
+ *
+ * Purpose:     Retrieve the requests in an event set.  Up to *count
+ *              requests are stored in the provided requests array, and
+ *              the connector ids corresponding to these requests are
+ *              stored in the provided connector_ids array.  Either or
+ *              both of these arrays may be NULL, in which case this
+ *              information is not returned.  If these arrays are
+ *              non-NULL, they must be large enough to contain *count
+ *              entries.  On exit, *count is set to the total number of
+ *              events in the event set.
+ *
+ *              Events are returned in the order they were added to the
+ *              event set.  If order is H5_ITER_INC or H5_ITER_NATIVE,
+ *              events will be returned starting from the oldest. If order
+ *              is H5_ITER_DEC, events will be returned starting with the
+ *              newest/most recent.
+ *
+ * Return:      SUCCEED / FAIL
+ *
+ * Programmer:  Neil Fortner
+ *              Tuesday, November 23, 2021
+ *
+ *-------------------------------------------------------------------------
+ */
+herr_t
+H5ESget_requests(hid_t es_id, H5_iter_order_t order, hid_t *connector_ids, void **requests, size_t array_len,
+                 size_t *count /*out*/)
```

Comment:
Looks fine, I think otherwise it's usually better to just name it `max_count`, similar to what `epoll_wait()` does: https://man7.org/linux/man-pages/man2/epoll_wait.2.html
That just makes it clearer that *count <= array_len

Reply:
OK forget my last comment, reading more I see it's kind of decoupled anyway and *count can be greater than array_len.